### PR TITLE
feat: add support for async toast

### LINKED ISSUE #3911: Loading toast status
# 🚀 Feature request

A new toast `status` called `loading`, and this one should display a spinner icon.

## 🧱 Problem Statement / Justification

Some toast libraries have a loading toast state, and it gives a great feedback for the user and the developer doesn't need to put the spinners directly in the screen and implementing loading conditions and this type of thing.

## ✅ Proposed solution or API

Shoud exist a new toast status:
```js
const toast = useToast()

toast({
  title: "Loading...",
  status: "loading", // | "info" | "warning" | "success" | "error"
  isClosable: false,
})
```

So, based on that property, the toast should display the spinner and a default background color, that can be light blue or other one that fits better.

## ↩️ Alternatives

Maybe if there's a way to attach the toast to a state, commonly `const [loading, setLoading] = useState(false)`, and then the toast would appear/disappear depending on the state. In the current implementation, only the duration is defined, and if this "trigger" based in a variable can be implemented I think would fit better, especially for this `loading` toast.

## 📝 Additional Information

It would be similar to this:
image
*from [Cogo Toast](https://github.com/Cogoport/cogo-toast)*

### PR DESCRIPTION
**Pull request by marek-sed**
Closes #3911 

## 📝 Description

> There is no support for async/loading toasts

## ⛳️ Current behavior (updates)

> No support for displaying loading status in toasts

## 🚀 New behavior

> Added toast.promise  as suggested in this discussion https://github.com/chakra-ui/chakra-ui/pull/4322#issuecomment-894024858

## 📝 Additional Information
this is a follow up of stale/closed PR https://github.com/chakra-ui/chakra-ui/pull/4322

### CODE REVIEW

**Comment by marek-sed** [comment]
Hi,

@timkindberg @segunadebayo , would be awesome if you find time to take a look.

**Comment by marek-sed** [comment]
Hello guys is there any interest to continue to pursue this feature?

**Comment by segunadebayo** [comment]
Hi @marek-sed,

We're definitely interested in getting this feature in. Thanks for all your work.

@TimKolberger, let's find a way to batch this into the upcoming major release. I think it'll be a good addition.

File: packages/toast/src/use-toast.tsx [hunk 1]
```diff
@@ -196,6 +200,24 @@ export function createStandaloneToast({
...
+    }: {
+      success: UseToastPromiseOption
+      error: UseToastPromiseOption
+      loading: UseToastPromiseOption
+    },
+  ) => {
+    const id = toastImpl({ ...loading, status: "loading", duration: null })
+    promise
+      .then(() => toastImpl.update(id, { ...success, status: "success" }))
+      .catch(() => toastImpl.update(id, { ...error, status: "error" }))
```
Comment (segunadebayo) [inline packages/toast/src/use-toast.tsx:218]:
```suggestion
      .then((data) => toastImpl.update(id, { ...success, status: "success", data }))
      .catch((error) => toastImpl.update(id, { ...error, status: "error", data: error }))
```

I think we need to provide the users will the resolved data or error so they can update the toast's message accordingly.

**Comment by Amit0617** [comment]
Hey did this got mentioned in chakra docs?

**Pull request merged**